### ISSUE TITLE
Add import functionality

### ISSUE THREAD

**Issue by ya332** [tldraw/tldraw#1828]
### What's the feature?

Can we have a feature to import files that we exported previously? 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by steveruizok** [comment]
Instead of opening them? We discussed this when we worked on the file export feature but decided to wait on this feature until "at least _someone_ asks for it". Congrats @ya332, you're that someone! 

A couple of questions, some of which you might answer, some of which we might have to answer:
- would you expect the whole saved file to be uploaded at once, or specific pages from the saved file?
- if there are two pages with the same name, would you expect the contents to be _merged_ into the existing page, or have the previous page _replaced_ by the importing page, or have a new page created so that we can keep both pages? (i.e. "My Page" and "My Page 2")
- if there are offline assets, we'd need to upload these, which might take some time and might even fail—would you expect a loading indicator? If the uploads fail, should we fail the whole import, or just those images?

**Comment by ya332** [comment]
I would expect whole file to be uploaded.
I think My page and My page 2 approach is common.
If offline assets fail, it is ok. I don't think whole import should fail.

Best
________________________________

**Comment by GodDoesNotPlayDice** [comment]
When this feature going to be added? D:

**Comment by xuanswe** [comment]
@steveruizok 

> would you expect the whole saved file to be uploaded at once, or specific pages from the saved file?

I expect to all pages by default, but it would be nice if there's a popup allow me to select which pages to include/exclude.

> if there are two pages with the same name, would you expect the contents to be merged into the existing page, or have the previous page replaced by the importing page, or have a new page created so that we can keep both pages? (i.e. "My Page" and "My Page 2")

I think it's better to keep both pages. But I think the name of new page could have the import time like this `My Page (2024-07-20T09:30:00Z)`

> if there are offline assets, we'd need to upload these, which might take some time and might even fail—would you expect a loading indicator? If the uploads fail, should we fail the whole import, or just those images?

I think offline assets failure should not prevent the whole import.
I expect to have a list of failures and be able to reupload each of them manually later.

**Comment by metamn** [comment]
For me this works perfect: https://github.com/tldraw/tldraw/issues/1650
